File: Algebra/Fundamentals/Greatest-Common-Divisor(GCD)/GreatestCommonDivisor.py
class GreatestCommonDivisor:
    def gcd(self, a, b):
        while b:
            a %= b
            a, b = b, a
        return a


ans = GreatestCommonDivisor()
res = ans.gcd(10, 8)
print("The gcd for given number is: ", res)

File: Algebra/Fundamentals/Greatest-Common-Divisor(GCD)/GreatestCommonDivisor.java
public class GreatestCommonDivisor {
    public static void swap(int a, int b){
        a = a - b;
        b = a + b;
        a = b - a;
    }

    public int gcd(int a, int b) {
        while (b > 0){
            a %= b;
            swap(a, b);
        }
        return a;
    }

    public static void main(String[] args) {
        int a = 10;
        int b = 8;
        GreatestCommonDivisor var = new GreatestCommonDivisor();
        int res = var.gcd(a, b);
        System.out.println("The required GCD for the given numer is: " + res);
    }
}
